### PR TITLE
Add categorical feature-based decision tree induction

### PR DESCRIPTION
**Pull request by semibran**
Quick test with `main_induction.py`. Training takes a couple of minutes or so but it automatically dumps and reloads model state after the first run so it's a bit easier to test and tweak.

Categorical features aren't a great indicator of claim acceptance so accuracy gets beat out by null model (for now). It's a decent start though and we can try adding dimensionality reduction, incorporating real-valued features, `sklearn.tree.DecisionTreeClassifier` etc.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+
 
 def run_menu(title, options):
     exit_menu = None
     while exit_menu is None:
         print(f'--- {title} ---')
-        input = OptionInput('Select option', options, lambda option: option[0])
-        _, option = input.get_input()
+        option_input = OptionInput('Select option', options, lambda option: option[0])
+        _, option = option_input.get_input()
```
Comment (semibran) [inline main.py:16]: pylint complains about overwrites since `input` is a builtin

File: main.py [hunk 2]
```diff
@@ -1,21 +1,21 @@
-import sys
-import time
```
Comment (semibran) [inline main.py:2]: unused?

**Review by jonathan-paugh — APPROVED** [review]
Very cool, looks good!
